build-test-deploy.yml

```diff
@@ -96,37 +96,77 @@ steps:
         $versionInfo = $(Get-Item .\GhAdSec\bin\x64\Release\AdSecGH.dll).VersionInfo
         echo $versionInfo
         $fullVersion = [array]${versionInfo}.FileVersion.split('.')
-        echo $fullVersion
         $currentTime = $(Get-Date -Format "dddd MM/dd/yyyy HH:mm")
-        echo $currentTime
         $majorVersion = [string]$fullVersion[0]
-        echo $majorVersion
         $minorVersion = [string]$fullVersion[1]
-        echo $minorVersion
         $spVersion = [string]$fullVersion[2]
-        echo $spVersion
         $buildNumber = [string]$fullVersion[3]
         echo $buildNumber
 
         $tag = [array]@($exeName,$majorVersion,$minorVersion,$spVersion,$buildNumber)
-        echo $tag
         $tagNumbersOnly = [array]@($majorVersion,$minorVersion,$spVersion,$buildNumber)
-        echo $tagNumbersOnly
 
         $dotSeparatedTag = $majorVersion + "." + $minorVersion + "." + $spVersion + "." + $buildNumber
-        echo $dotSeparatedTag
         $underscoreSeparatedTag = $majorVersion + "_" + $minorVersion + "_" + $spVersion + "_" + $buildNumber
-        echo $underscoreSeparatedTag
         $tagNumbersOnlyUnderscore = $majorVersion + "_" + $minorVersion + "_" + $spVersion + "_" + $buildNumber
-        echo $tagNumbersOnlyUnderscore
         $tagNumbersOnlydot = $majorVersion + "." + $minorVersion + "." + $spVersion + "." + $buildNumber
-        echo $tagNumbersOnlydot
 
         git rev-parse $underscoreSeparatedTag
         if($lastExitCode -eq 0){Write-Host ("##vso[task.setvariable variable=TAG_EXISTS]true")}
         if($lastExitCode -eq 0){
           $tagExists = 'true'
+          echo $tagExists
         }
     errorActionPreference: 'continue'
     workingDirectory: '$(System.DefaultWorkingDirectory)'
-    ignoreLASTEXITCODE: true
\ No newline at end of file
+    ignoreLASTEXITCODE: true
+
+- task: PowerShell@2
+  inputs:
+    targetType: 'inline'
+    script: |
+         $xml = [Xml] (Get-Content GhAdSec/AdSecGH.csproj)
+         $version = $xml.Project.PropertyGroup.Version
+         echo $version
+         echo "##vso[task.setvariable variable=version]$version"
+
+- task: NuGetCommand@2
+  inputs:
+    command: 'pack'
+    packagesToPack: 'GhAdSec/*.csproj'
+    versioningScheme: 'byEnvVar'
+    versionEnvVar: 'version'
+  condition: succeeded()
+  #condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/main'))
+  displayName: Package NuGet
+
+- task: GitHubRelease@1
+  inputs:
+    gitHubConnection: 'github.com_peterjgrainger'
+    repositoryName: 'arup-group/AdSec-Grasshopper'
+    action: 'edit'
+    target: '$(Build.SourceVersion)'
+    tag: '$(version)'
+    title: 'AdSecGH Version $dotSeparatedTag'
```

Comment:
@peterjgrainger for some reason this is not picked up here correctly..

Reply:
I'll try to fix this :)